test/run/object-extend-type.mo

```diff
@@ -20,3 +20,14 @@ let bm = { b with M };
 
 let aM = { a and M };
 let bM = { b and M };
+
+// poor man's type definition to declare object type with type fields
```

Comment:
This is clever, I didn't imagine this is possible at all with type fields! Now I can implement the commented test above :-)

Reply:
We should just add type fields proberly (to record types and values), but for now the hack works.